src/adaptors/adaptor_tls.c

```diff
@@ -0,0 +1,630 @@
+            } else {
+                res = pn_tls_config_set_peer_authentication(tls->tls_config, PN_TLS_ANONYMOUS_PEER, 0);
+            }
+        } else {
+            // Connector.
+            if (config->verify_host_name) {
+                res = pn_tls_config_set_peer_authentication(
+                    tls->tls_config, PN_TLS_VERIFY_PEER_NAME, config_ssl_profile->ssl_trusted_certificate_db);
+            } else {
+                res = pn_tls_config_set_peer_authentication(
+                    tls->tls_config, PN_TLS_VERIFY_PEER, config_ssl_profile->ssl_trusted_certificate_db);
+            }
+
+            // intitalize tls_has_output to true when the router is acting as the client initiating TLS handshake.
+            tls->tls_has_output = true;
+        }
+
+        if (res != 0) {
+            qd_log(tls->log_source,
+                   QD_LOG_ERROR,
+                   "[C%" PRIu64 "] Unable to set tls peer authentication for sslProfile %s - (%d)",
+                   tls->conn_id,
+                   config->ssl_profile_name,
+                   res);
+            break;
+        }
+
+        //
+        // Provide an ordered list of application protocols for ALPN by calling pn_tls_config_set_alpn_protocols. In our
+        // case, h2 is the only supported protocol. A list of protocols can be found here -
+        // https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.txt
+        //
+        if (alpn_protocols) {
+            pn_tls_config_set_alpn_protocols(tls->tls_config, alpn_protocols, alpn_protocol_count);
+        }
+
+        // set up tls session
+        if (tls->tls_session) {
+            pn_tls_free(tls->tls_session);
+        }
+        tls->tls_session = pn_tls(tls->tls_config);
+
+        if (!tls->tls_session) {
+            qd_log(tls->log_source,
+                   QD_LOG_ERROR,
+                   "[C%" PRIu64 "] Unable to create tls session for sslProfile %s with hostname: '%s'",
+                   tls->conn_id,
+                   config->ssl_profile_name,
+                   config->host);
+            break;
+        }
+
+        pn_tls_set_peer_hostname(tls->tls_session, config->host);
+
+        int ret = pn_tls_start(tls->tls_session);
+        if (ret != 0) {
+            break;
+        }
+
+        qd_log(tls->log_source,
+               QD_LOG_INFO,
+               "[C%" PRIu64 "] Successfully configured sslProfile %s",
+               tls->conn_id,
+               config->ssl_profile_name);
+        return true;
+
+    } while (0);
+
+    // Handle tls creation/setup failure by deleting any pn domain or session objects
+    if (tls->tls_session) {
+        pn_tls_free(tls->tls_session);
+        tls->tls_session = 0;
+    }
+
+    if (tls->tls_config) {
+        pn_tls_config_free(tls->tls_config);
+        tls->tls_config = 0;
+    }
+
+    return false;
+}
+
+void set_qdr_connection_info_details(qd_tls_t *tls, qdr_connection_info_t *conn_info)
```

Comment:
This function is meant to be called by all adaptors right after the handshake is complete so that it can set the connection info details. This function could go in adaptor_common.c but since it accepts qd_tls_t as one of its parameters, I chose to keep this fuction here. When I raise my next PR with the TCP TLS implementation, you will see how this function is called and based on that we can make a decision where it should live but for now I want to keep it here.

Reply:
OK